### PR TITLE
feat: archive/unarchive notes

### PR DESCRIPTION
**Pull request by himadri2110**
Features: 

- Archive note
- Unarchive note
- Edit note in Archives (Modified ArchiveController & server.js)
- Updated noteState to store both notes and archives
- Updated notesReducer
- Created Archive Context

### CODE REVIEW

File: src/components/NoteCard/NoteCard.jsx [hunk 1]
```diff
@@ -4,16 +4,25 @@ import {
   ColorLensOutlinedIcon,
   LabelOutlinedIcon,
   ArchiveOutlinedIcon,
+  UnarchiveOutlinedIcon,
   DeleteOutlinedIcon,
   PushPinIcon,
   PushPinOutlinedIcon,
 } from "assets/index";
-import { useNotes } from "contexts";
+import { useArchive, useNotes } from "contexts";
```
Comment (vijaytembugade) [inline src/components/NoteCard/NoteCard.jsx:12]: Good set of icons!

**Review by vijaytembugade — APPROVED** [review]
Everything looks clean. Good work

File: src/components/Sidebar/Sidebar.jsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ const Sidebar = () => {
         <ul>
           <li>
             <NavLink
-              to="/home"
+              to="/"
               className={({ isActive }) => (isActive ? "active" : undefined)}
             >
               <i className="fa-solid fa-house"></i> Home
```
Comment (pritam-kr) [inline src/components/Sidebar/Sidebar.jsx:17]: is it working font awesome ?/

File: src/contexts/archiveContext.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+      if (status === 201) {
+        dispatchNote({
+          type: "SET_NOTES_AND_ARCHIVE",
+          payload: { notes: data.notes, archives: data.archives },
+        });
+      }
+    } catch (err) {
+      console.error(err);
+    }
```
Comment (pritam-kr) [inline src/contexts/archiveContext.js:41]: suggestion: don't handle your error with the console. you can use any other tool like toast na

File: src/reducers/noteReducer.js [hunk 4]
```diff
@@ -1,7 +1,11 @@
 const noteReducer = (state, { type, payload }) => {
   switch (type) {
-    case "GET_NOTES":
-      return [...payload];
+    case "SET_NOTES":
+      return { ...state, notes: payload };
+    case "SET_ARCHIVED":
+      return { ...state, archives: payload };
+    case "SET_NOTES_AND_ARCHIVE":
+      return { ...state, notes: payload.notes, archives: payload.archives };
```
Comment (pritam-kr) [inline src/reducers/noteReducer.js:8]: Again, Suggestions. your code is so clean. please make a const variable for it too.(actions)

**Review by pritam-kr — APPROVED** [review]
Everything is good himdari. 🔥
- I like the way, your fetch functions are on another page. 
- I didn't find any issue as per my knowledge.

File: src/contexts/archiveContext.js (hunk 3 above)
Comment (himadri2110) [reply src/contexts/archiveContext.js:41]: Yes, I'll add toasts later. Thanks.

File: src/reducers/noteReducer.js (hunk 4 above)
Comment (himadri2110) [reply src/reducers/noteReducer.js:8]: Yes, I'll do that. Thanks.

**Review by poornima-alugubelly — APPROVED** [review]
Code looks clean nothing to change 👍 good work

**Pull request merged**